### PR TITLE
ddl_sink (ticdc): fix ddl sink nil point panic

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #4934 

#### Tests 

 - Integration test
 - Manual test (add detailed scripts or steps below)
1. Add a `time.Sleep(time.Second * 10)` above: https://github.com/pingcap/tiflow/blob/master/cdc/owner/ddl_sink.go#L111
2. make cdc 
3. start cdc server
4. create a changefeed with syncpoint enable
5. no panic 
6. pause changefeed and update it, disable syncpoint , rusume this changefeed
7. no panic 

### Release note 

```release-note
Fix ddl sink panic when changefeed syncpoint is enable.
```

### CODE REVIEW

**Review by nongfushanquan — APPROVED** [review]

File: cdc/owner/changefeed.go [hunk 1]
```diff
@@ -263,7 +263,11 @@ func (c *changefeed) tick(ctx cdcContext.Context, state *orchestrator.Changefeed
 		return errors.Trace(err)
 	default:
 	}
-
+	// we need to wait sink to be ready before we do the other things
+	// otherwise, we may cause a nil pointer panic
+	if !c.sink.isInitialized() {
```
Comment (3AceShowHand) [inline cdc/owner/changefeed.go:268]: how about move this inside the sink ?

**Review by 3AceShowHand — APPROVED** [review]

**Pull request merged**